src/cards/impending_doom.ts

```diff
@@ -0,0 +1,106 @@
+import { IUnit } from '../entity/Unit';
+import * as Image from '../graphics/Image';
+import { Spell } from './index';
+import * as Unit from '../entity/Unit';
+import Underworld from '../Underworld';
+import { CardCategory } from '../types/commonTypes';
+import { playDefaultSpellAnimation, playDefaultSpellSFX } from './cardUtils';
+import floatingText from '../graphics/FloatingText';
+import * as colors from '../graphics/ui/colors';
+import { CardRarity, probabilityMap } from '../types/commonTypes';
+import { getOrInitModifier } from './util';
+
+export const impendingDoomCardId = 'impending doom';
+function add(unit: Unit.IUnit, underworld: Underworld, prediction: boolean, quantity: number = 1) {
+  const modifier = getOrInitModifier(unit, impendingDoomCardId, { isCurse: true, quantity, persistBetweenLevels: false }, () => {
+    // Add event
+    if (!unit.onTurnEndEvents.includes(impendingDoomCardId)) {
+      unit.onTurnEndEvents.push(impendingDoomCardId);
+    }
+    // Add subsprite image
+    if (!prediction) {
+      if (spell.modifiers?.subsprite) {
+        Image.addSubSprite(unit.image, spell.modifiers.subsprite.imageName);
+      }
+    }
+  });
+
+  if (!prediction) {
+    // Temporarily use floating text until spell animation is finished
+    floatingText({ coords: unit, text: impendingDoomCardId });
+    updateTooltip(unit);
+  }
+}
+export function updateTooltip(unit: Unit.IUnit) {
+  if (unit.modifiers[impendingDoomCardId]) {
+    // Set tooltip:
+    unit.modifiers[impendingDoomCardId].tooltip = `Impending doom ${unit.modifiers[impendingDoomCardId].quantity}...`
+  }
+}
+
+//Dev spell for toxic res: Unit dies in specified number of turns
```

Comment:
Is this just a copy of suffocate for toxic res? If so, it doesn't need to be a whole spell, it can just be a modifier.  See src/modifierSummoningSickness and src/cards/index.ts line 272 (registerSummoningSickness()) and priest.ts line 41 for how to use/create modifiers that aren't attached to a card